compiler/src/dotty/tools/dotc/classpath/ClassPathFactory.scala

```diff
@@ -52,14 +53,30 @@ class ClassPathFactory {
 
   // Internal
   protected def classesInPathImpl(path: String, expand: Boolean)(using Context): List[ClassPath] =
```

Comment:
if I had one comment for cleanup - there should probably be a third argument i.e. `expandManifest: Boolean` and this would only be true specifically when calling `classesInExpandedPath(javaUserClassPath)` in PathResolvers

Reply:
which then isolates the expansion specifically to the `java.class.path` property

Reply:
and then finally move the logic to `expandPath`?

Reply:
all not blockers